lib/identity/client.js

```diff
@@ -141,6 +139,35 @@ class Client extends PartialClient {
     const resultQuery = await this.storage.search(request)
     return resultQuery
   }
+  async updateSecret(secret) {
+    // Search for all records of the same type
+    const request = new Search(true)
+    let searchType = `tozny.secret.${SECRET_UUID}.${secret.secretType}.${secret.secretName}`
+    request.match({ type: searchType })
+    const resultQuery = await this.storage.search(request)
+    const secretList = await resultQuery.next()
+    // Find the Record Tagged Latest
+    let recordId
+    for (var i = 0; i < secretList.length; i++) {
+      if (secretList[i].meta.plain['version'] == 'latest') {
+        recordId = secretList[i].meta.recordId
+      }
+    }
+    // Update latest Record to be new version number
+    const versionNumber = resultQuery.totalResults
```

Comment:
This is assuming too much, what if someone has deleted one of the secret versions, then the count of results will not equal the version. Instead you should get the version number based on what the version number is for the secret/record tagged “latest”